### PR TITLE
[model] feat: support NPU wan2.2_I2V_5B 

### PR DESCRIPTION
**Pull request by dzy00897185**
### What does this PR do?
Adds Ascend NPU support for wan2.2_I2V_5B training. First, training-related YAML and JSON configuration files are added under VeOmni/configs/dit. Based on the original T2V training codebase, modifications such as additional input channels are implemented to accommodate the new I2V logic. This iteration also lays groundwork for future integration of the I2V-14B model.

### Checklist Before Starting
- Search for relative PRs/issues and link here:
- PR title follows `[{modules}] {type}: {description}` format (enforced by [check_pr_title.yml](.github/workflows/check_pr_title.yml))
  - **Allowed modules:** `agent`, `ci`, `ckpt`, `config`, `data`, `dist`, `docker`, `docs`, `logging`, `lora`, `misc`, `model`, `omni`, `optim`, `ops`, `parallel`, `perf`, `release`, `task`, `trainer`
  - **Allowed types:** `feat`, `fix`, `refactor`, `chore`, `test`
  - Breaking changes: prepend `[BREAKING]` — e.g. `[BREAKING][parallel, model] feat: dynamic batching`

### Test

> Validation results (training curves, eval metrics) for changes not covered by CI.

### API and Usage Example

> Show API changes and usage examples if applicable.

### Design & Code Changes
configs/dit/wan2.2_I2V_5B.yaml.yaml
configs/model_configs/wan/wan2.2_i2v_5b.json
veomni/models/diffusers/wan_t2v/wan_condition/configuration_wan_condition.py
veomni/models/diffusers/wan_t2v/wan_condition/modeling_wan_condition.py
veomni/models/diffusers/wan_t2v/wan_transformer/configuration_wan_transformer.py
veomni/models/diffusers/wan_t2v/wan_transformer/modeling_wan_transformer.py

> High-level design description and specific change list.

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks
- Added/updated documentation
- If `tasks/` training scripts were moved or renamed: updated `docs/` examples and verified `python3 scripts/ci/check_doc_task_paths.py` passes (also enforced by the **Check doc task paths** CI workflow)
- Added tests to CI workflow (or explained why not feasible)

## Summary by CodeRabbit

- **New Features**
  - Added support for Wan 2.2 Image-to-Video 5B training.
  - Added configurable image conditioning, masks, latent inputs, and expanded timestep handling.
  - Added model configuration for attention, positional encoding, patching, and text embeddings.
  - Improved image-to-video loss handling and support for 36- and 48-channel inputs.

- **Bug Fixes**
  - Improved compatibility with gradient checkpointing and mixed-precision attention.
  - Added safer handling for missing or mismatched configuration values and prediction targets.

### CODE REVIEW

File: configs/dit/wan2.2_I2V_5B.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  ops_implementation:
+    rotary_pos_emb_implementation: eager
+    rms_norm_implementation: eager
+    swiglu_mlp_implementation: eager
+    cross_entropy_loss_implementation: eager
+    load_balancing_loss_implementation: eager
+    moe_implementation: eager
+
+data:
+  train_path: /home/d00897185/Tom-and-Jerry-VideoGeneration-Dataset-parquet
```
[the remark this answers is not in the document]
Comment (dzy00897185) [reply configs/dit/wan2.2_I2V_5B.yaml]: done

File: veomni/models/diffusers/wan_t2v/wan_transformer/modeling_wan_transformer.py [hunk 2]
```diff
@@ -259,31 +253,103 @@ def apply_rotary_emb(
...
 
 
-# ================================================================
+# =================================================================
+# Gradient-checkpointing wrapper for WanTransformerBlock.
+# VeOmni's _gradient_checkpointing_func passes an `early_stop` kwarg
+# that diffusers' WanTransformerBlock.forward() does not accept.
+# This wrapper absorbs and ignores it.
+# =================================================================
+def _wan_block_ckpt_wrapper(block):
```
[the remark this answers is not in the document]
Comment (dzy00897185) [reply veomni/models/diffusers/wan_t2v/wan_transformer/modeling_wan_transformer.py:276]: done

File: veomni/models/diffusers/wan_t2v/wan_transformer/modeling_wan_transformer.py [hunk 3]
```diff
@@ -413,24 +487,98 @@ def config(self, value):
...
+                timestep=ts,
+                encoder_hidden_states=enc_hs,
+                encoder_hidden_states_image=encoder_hidden_states_image,
+                image_latents=img_lat,
+                mask=msk,
             )
             predictions.append(prediction)
+
+            if prediction.shape != target.shape:
+                min_t = min(prediction.shape[2], target.shape[2])
```
[the remark this answers is not in the document]
Comment (dzy00897185) [reply veomni/models/diffusers/wan_t2v/wan_transformer/modeling_wan_transformer.py:572]: done

File: veomni/models/diffusers/wan_t2v/wan_transformer/modeling_wan_transformer.py [hunk 4]
```diff
@@ -413,24 +487,98 @@ def config(self, value):
...
+                    norm_lat = latents[i]
+                    if norm_lat is None:
+                        raise ValueError(
+                            "norm_latents is required for I2V training when image_latents is not provided"
+                        )
+                    img_lat = norm_lat[:, :, 0:1, :, :]  # [B, 16, 1, H, W]
+                    img_lat_source = "from_norm_latents"
+                    if get_parallel_state().dp_rank == 0:
+                        logger.info(
+                            f"[I2V image_latents] source={img_lat_source}, "
```
[the remark this answers is not in the document]
Comment (dzy00897185) [reply veomni/models/diffusers/wan_t2v/wan_transformer/modeling_wan_transformer.py:543]: done

File: configs/dit/wan2.2_I2V_5B.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    fsdp_config:
+      fsdp_mode: fsdp2
+  mixed_precision:
+    enable: true
+    param_dtype: bfloat16
+    reduce_dtype: bfloat16
+    buffer_dtype: bfloat16
+  checkpoint:
+    output_dir: Wan2.2-I2V-5B
+  train.checkpoint:
```
[the remark this answers is not in the document]
Comment (dzy00897185) [reply configs/dit/wan2.2_I2V_5B.yaml:42]: done

**Comment by FoolPlayer** [comment]
Hi, thanks for you PR, please fix CI.

File: veomni/distributed/torch_parallelize.py [hunk 6]
```diff
@@ -595,6 +595,13 @@ def build_parallelize_model(
         if not use_reentrant:
             gradient_checkpointing_kwargs["early_stop"] = checkpoint_early_stop
 
+        try:
+            from torch.utils.checkpoint import _CheckpointFrame
+            _CheckpointFrame.check_recomputed_tensors_match = lambda self, *a, **kw: None
```
Comment (Crystal-jiang) [inline veomni/distributed/torch_parallelize.py:644]: This globally patches `_CheckpointFrame.check_recomputed_tensors_match` to a no-op for **all models**, not just Wan2.2 on NPU. This disables PyTorch's safety check that catches dtype/shape mismatches during gradient checkpoint recomputation. If another model has a genuine recomputation bug, it will silently produce wrong gradients with no warning.

Suggestions:
- Scope this to NPU-only, or add a comment explaining why the check must be disabled on NPU.
- The `except (ImportError, AttributeError): pass` silently swallows failures. If a future torch version renames `_CheckpointFrame`, the patch is silently inactive. Consider logging a warning.

File: configs/dit/wan2.2_I2V_5B.yaml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
```
Comment (Crystal-jiang) [inline configs/dit/wan2.2_I2V_5B.yaml:42]: `train.checkpoint:` is a literal dotted key nested inside the `train:` block, so it resolves to `train["train.checkpoint"]` instead of merging into the `checkpoint:` block at line 41. This means `save_hf_weights: true` is likely silently ignored.

Compare with `configs/dit/wan_sft.yaml` where `save_hf_weights` is correctly placed under `checkpoint:`:

```yaml
  checkpoint:
    output_dir: wan2.1-i2v-14b-480p
    manager: dcp
    save_hf_weights: true
```

Move `save_hf_weights` under the existing `checkpoint:` block.

File: veomni/models/diffusers/wan_t2v/wan_transformer/modeling_wan_transformer.py [hunk 8]
```diff
@@ -413,24 +488,98 @@ def config(self, value):
...
+                hidden_states=hs,
+                timestep=ts,
+                encoder_hidden_states=enc_hs,
+                encoder_hidden_states_image=encoder_hidden_states_image,
+                image_latents=img_lat,
+                mask=msk,
             )
             predictions.append(prediction)
+
+            if prediction.shape != target.shape:
```
Comment (Crystal-jiang) [inline veomni/models/diffusers/wan_t2v/wan_transformer/modeling_wan_transformer.py:571]: Silently cropping `prediction` and `target` to their common extent converts a shape bug into a quietly-wrong loss: gradients are computed on a sub-volume, and the discarded region is never supervised. This is especially risky because the config declares `out_channels: 48` while the training target comes from 16-channel latents.

If shapes cannot legitimately differ, use an assertion so the mismatch surfaces immediately:

```python
assert prediction.shape == target.shape, f"Shape mismatch: {prediction.shape} vs {target.shape}"
```

File: configs/dit/wan2.2_I2V_5B.yaml [hunk 9]
```diff
@@ -0,0 +1,55 @@
+model:
+  model_path: Wan-AI/Wan2___2-TI2V-5B-Diffusers/transformer
+  condition_model_path: Wan-AI/Wan2___2-TI2V-5B-Diffusers
+  config_path: VeOmni/configs/model_configs/wan/wan2.2_i2v_5b.json
```
Comment (Crystal-jiang) [inline configs/dit/wan2.2_I2V_5B.yaml:4]: `config_path: VeOmni/configs/...` requires CWD to be one level above the repo root to resolve. Other configs in `configs/dit/` use repo-relative paths (e.g. `wan_sft.yaml` uses `./configs/model_configs/wan/wani2v_14b.json`).

Use `configs/model_configs/wan/wan2.2_i2v_5b.json` for portability.

**Review by Crystal-jiang — COMMENTED** [review]
Code review feedback. See inline comments for details.
